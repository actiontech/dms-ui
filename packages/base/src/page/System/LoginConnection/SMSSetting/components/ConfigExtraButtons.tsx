import { useTranslation } from 'react-i18next';
import { Space, Form, message } from 'antd';
import { BasicInput } from '@actiontech/dms-kit';
import { FormItemLabel } from '@actiontech/dms-kit';
import { useRef, useState } from 'react';
import baseConfiguration from '@actiontech/shared/lib/api/base/service/Configuration';
import { ResponseCode } from '@actiontech/dms-kit';
import { formItemLayout } from '@actiontech/dms-kit/es/components/CustomForm/style';
import {
  ConfigModifyBtn,
  ConfigTestBtn,
  ConfigTestPopoverForm
} from '@actiontech/dms-kit';
export interface ConfigExtraButtonsProps {
  isConfigClosed: boolean;
  extraButtonsVisible: boolean;
  handleClickModify: () => void;
  enabled: boolean;
}
const ConfigExtraButtons = ({
  isConfigClosed,
  extraButtonsVisible,
  handleClickModify,
  enabled
}: ConfigExtraButtonsProps) => {
  const { t } = useTranslation();
  const [messageApi, messageContextHolder] = message.useMessage();
  const [testForm] = Form.useForm<{
    phone: string;
  }>();
  const testing = useRef(false);
  const [testPopoverVisible, toggleTestPopoverVisible] = useState(false);
  const testCodingConfiguration = async () => {
    if (testing.current) {
      return;
    }
    const values = await testForm.validateFields();
    testing.current = true;
    toggleTestPopoverVisible(false);
    const hide = messageApi.loading(
      t('dmsSystem.global.smsSetting.testing'),
      0
    );
    baseConfiguration
      .TestSmsConfiguration({
        test_sms_configuration: {
          recipient_phone: values.phone
        }
      })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          if (res.data.data?.is_smtp_send_normal) {
            messageApi.success(t('dmsSystem.global.smsSetting.testSuccess'));
          } else {
            messageApi.error(
              res.data.data?.send_error_message ?? t('common.unknownError')
            );
          }
        }
      })
      .finally(() => {
        hide();
        testing.current = false;
        testForm.resetFields();
      });
  };
  const onTestPopoverOpen = (open: boolean) => {
    if (!enabled) {
      return;
    }
    if (!open) {
      testForm.resetFields();
    }
    toggleTestPopoverVisible(open);
  };
  return (
    <>
      {messageContextHolder}
      <Space size={12} hidden={isConfigClosed || !extraButtonsVisible}>
        <ConfigTestBtn
          testingRef={testing}
          popoverOpen={testPopoverVisible}
          onPopoverOpenChange={onTestPopoverOpen}
          popoverForm={
            <ConfigTestPopoverForm
              handleTest={testCodingConfiguration}
              handleCancel={() => {
                onTestPopoverOpen(false);
              }}
            >
              <Form form={testForm} colon={false} {...formItemLayout.fullLine}>
                <FormItemLabel
                  name="phone"
                  label={t('common.phone')}
                  style={{
                    marginBottom: 0
                  }}
                  rules={[
                    {
                      required: true
                    }
                  ]}
                >
                  <BasicInput
                    placeholder={t('common.form.placeholder.input', {
                      name: t('common.phone')
                    })}
                  />
                </FormItemLabel>
              </Form>
            </ConfigTestPopoverForm>
          }
        />
        <ConfigModifyBtn
          className="modify-coding-configuration"
          onClick={handleClickModify}
        />
      </Space>
    </>
  );
};
export default ConfigExtraButtons;
