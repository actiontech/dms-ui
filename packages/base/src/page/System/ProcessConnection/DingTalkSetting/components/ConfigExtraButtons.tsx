import { useTranslation } from 'react-i18next';
import { useRef } from 'react';

import { message, Space } from 'antd';
import { BasicToolTips, BasicButton } from '@actiontech/shared';
import ConfigModifyBtn from '../../../components/ConfigModifyBtn';

import configuration from '@actiontech/shared/lib/api/sqle/service/configuration';
import { ResponseCode } from '@actiontech/shared/lib/enum';
import { ThunderBoltFilled } from '@actiontech/icons';

export interface ConfigExtraButtonsProps {
  isConfigClosed: boolean;
  extraButtonsVisible: boolean;
  handleClickModify: () => void;
}

const ConfigExtraButtons = ({
  isConfigClosed,
  extraButtonsVisible,
  handleClickModify
}: ConfigExtraButtonsProps) => {
  const { t } = useTranslation();
  const [messageApi, messageContextHolder] = message.useMessage();

  const testTing = useRef(false);
  const testDingTalkConfiguration = () => {
    if (testTing.current) {
      return;
    }
    testTing.current = true;
    configuration
      .testDingTalkConfigV1()
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          if (res.data.data?.is_ding_talk_send_normal) {
            messageApi.success(t('dmsSystem.dingTalk.testSuccess'));
          } else {
            messageApi.error(
              res.data.data?.send_error_message ?? t('common.unknownError')
            );
          }
        }
      })
      .finally(() => {
        testTing.current = false;
      });
  };

  return (
    <>
      {messageContextHolder}
      <Space size={12} hidden={isConfigClosed || !extraButtonsVisible}>
        <BasicToolTips title={t('common.test')} titleWidth={54}>
          <BasicButton
            type="text"
            className="system-config-button"
            loading={testTing.current}
            disabled={testTing.current}
            icon={<ThunderBoltFilled />}
            onClick={testDingTalkConfiguration}
          />
        </BasicToolTips>
        <ConfigModifyBtn onClick={handleClickModify} />
      </Space>
    </>
  );
};

export default ConfigExtraButtons;
