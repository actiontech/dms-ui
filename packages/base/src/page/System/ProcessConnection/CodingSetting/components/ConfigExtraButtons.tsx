import { useTranslation } from 'react-i18next';
import { Space, Form, message } from 'antd';
import {
  ConfigTestBtn,
  ConfigTestPopoverForm,
  BasicInput,
  ConfigModifyBtn
} from '@actiontech/shared';
import { FormItemLabel } from '@actiontech/shared/lib/components/CustomForm';
import { useRef, useState } from 'react';
import configuration from '@actiontech/shared/lib/api/sqle/service/configuration';
import { ResponseCode } from '@actiontech/shared/lib/enum';
import { formItemLayout } from '@actiontech/shared/lib/components/CustomForm/style';

export interface ConfigExtraButtonsProps {
  isConfigClosed: boolean;
  extraButtonsVisible: boolean;
  handleClickModify: () => void;
  enabled: boolean;
}

const ConfigExtraButtons = ({
  isConfigClosed,
  extraButtonsVisible,
  handleClickModify,
  enabled
}: ConfigExtraButtonsProps) => {
  const { t } = useTranslation();

  const [messageApi, messageContextHolder] = message.useMessage();

  const [testForm] = Form.useForm<any>();

  const testing = useRef(false);
  const [testPopoverVisible, toggleTestPopoverVisible] = useState(false);

  const testCodingConfiguration = async () => {
    if (testing.current) {
      return;
    }
    const values = await testForm.validateFields();

    testing.current = true;
    toggleTestPopoverVisible(false);
    const hide = messageApi.loading(t('dmsSystem.codingDocking.testing'), 0);
    configuration
      .testCodingConfigV1({ coding_project_name: values.codingProjectName })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          if (res.data.data?.is_message_sent_normally) {
            messageApi.success(t('dmsSystem.codingDocking.testSuccess'));
          } else {
            messageApi.error(
              res.data.data?.error_message ?? t('common.unknownError')
            );
          }
        }
      })
      .finally(() => {
        hide();
        testing.current = false;
        testForm.resetFields();
      });
  };

  const onTestPopoverOpen = (open: boolean) => {
    if (!enabled) {
      return;
    }
    if (!open) {
      testForm.resetFields();
    }
    toggleTestPopoverVisible(open);
  };

  return (
    <>
      {messageContextHolder}
      <Space size={12} hidden={isConfigClosed || !extraButtonsVisible}>
        <ConfigTestBtn
          testingRef={testing}
          popoverOpen={testPopoverVisible}
          onPopoverOpenChange={onTestPopoverOpen}
          popoverForm={
            <ConfigTestPopoverForm
              handleTest={testCodingConfiguration}
              handleCancel={() => {
                onTestPopoverOpen(false);
              }}
            >
              <Form form={testForm} colon={false} {...formItemLayout.fullLine}>
                <FormItemLabel
                  name="codingProjectName"
                  label={t('dmsSystem.codingDocking.codingProjectName')}
                  style={{ marginBottom: 0 }}
                  rules={[{ required: true }]}
                >
                  <BasicInput
                    placeholder={t('common.form.placeholder.input', {
                      name: t('dmsSystem.codingDocking.codingProjectName')
                    })}
                  />
                </FormItemLabel>
              </Form>
            </ConfigTestPopoverForm>
          }
        />
        <ConfigModifyBtn
          className="modify-coding-configuration"
          onClick={handleClickModify}
        />
      </Space>
    </>
  );
};

export default ConfigExtraButtons;
