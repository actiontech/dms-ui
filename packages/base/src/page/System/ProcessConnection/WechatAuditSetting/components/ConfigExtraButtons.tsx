import { useTranslation } from 'react-i18next';
import { useRef, useState } from 'react';
import { Form, message, Space } from 'antd';
import { BasicInput } from '@actiontech/shared';
import ConfigModifyBtn from '../../../components/ConfigModifyBtn';
import configuration from '@actiontech/shared/lib/api/sqle/service/configuration';
import { ResponseCode } from '@actiontech/shared/lib/enum';
import ConfigTestBtn from '../../../components/ConfigTestBtn';
import ConfigTestPopoverForm from '../../../components/ConfigTestPopoverForm';
import { TestFormFields } from '../index.type';
import { formItemLayout } from '@actiontech/shared/lib/components/FormCom/style';
import { FormItemLabel } from '@actiontech/shared/lib/components/FormCom';

export interface ConfigExtraButtonsProps {
  isConfigClosed: boolean;
  extraButtonsVisible: boolean;
  handleClickModify: () => void;
  enabled: boolean;
}

const ConfigExtraButtons = ({
  isConfigClosed,
  extraButtonsVisible,
  handleClickModify,
  enabled
}: ConfigExtraButtonsProps) => {
  const { t } = useTranslation();
  const [messageApi, messageContextHolder] = message.useMessage();
  const [testForm] = Form.useForm<TestFormFields>();

  const testing = useRef(false);
  const [testPopoverVisible, toggleTestPopoverVisible] = useState(false);

  const testWechatAuditConfiguration = async () => {
    if (testing.current) {
      return;
    }
    const values = await testForm.validateFields();

    testing.current = true;
    toggleTestPopoverVisible(false);
    const hide = messageApi.loading(t('dmsSystem.wechatAudit.testing'), 0);

    configuration
      .testWechatAuditConfigV1({ wechat_id: values.wechatUserID })
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          if (res.data.data?.is_message_sent_normally) {
            messageApi.success(t('dmsSystem.wechatAudit.testSuccess'));
          } else {
            messageApi.error(
              res.data.data?.error_message ?? t('common.unknownError')
            );
          }
        }
      })
      .finally(() => {
        hide();
        testing.current = false;
        testForm.resetFields();
      });
  };

  const onTestPopoverOpen = (open: boolean) => {
    if (!enabled) {
      return;
    }
    if (!open) {
      testForm.resetFields();
    }
    toggleTestPopoverVisible(open);
  };

  return (
    <>
      {messageContextHolder}
      <Space size={12} hidden={isConfigClosed || !extraButtonsVisible}>
        <ConfigTestBtn
          testingRef={testing}
          popoverOpen={testPopoverVisible}
          onPopoverOpenChange={onTestPopoverOpen}
          popoverForm={
            <ConfigTestPopoverForm
              handleTest={testWechatAuditConfiguration}
              handleCancel={() => {
                onTestPopoverOpen(false);
              }}
            >
              <Form form={testForm} colon={false} {...formItemLayout.fullLine}>
                <FormItemLabel
                  name="wechatUserID"
                  label={t('dmsSystem.wechatAudit.wechatUserID')}
                  style={{ marginBottom: 0 }}
                  rules={[{ required: true }]}
                >
                  <BasicInput
                    placeholder={t('common.form.placeholder.input', {
                      name: t('dmsSystem.wechatAudit.wechatUserID')
                    })}
                  />
                </FormItemLabel>
              </Form>
            </ConfigTestPopoverForm>
          }
        />
        <ConfigModifyBtn onClick={handleClickModify} />
      </Space>
    </>
  );
};

export default ConfigExtraButtons;
