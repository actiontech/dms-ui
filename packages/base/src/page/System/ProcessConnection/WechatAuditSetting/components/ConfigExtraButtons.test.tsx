import ConfigExtraButtons, {
  ConfigExtraButtonsProps
} from './ConfigExtraButtons';
import { cleanup, fireEvent, act, screen } from '@testing-library/react';
import { superRender } from '@actiontech/shared/lib/testUtil/superRender';
import {
  getAllBySelector,
  getBySelector
} from '@actiontech/shared/lib/testUtil/customQuery';
import system from '../../../../../testUtils/mockApi/system';
import { createSpySuccessResponse } from '@actiontech/shared/lib/testUtil/mockApi';
import { mockUseCurrentUser } from '@actiontech/shared/lib/testUtil/mockHook/mockUseCurrentUser';

describe('base/System/ProcessConnection/WechatAuditSetting/ConfigExtraButtons', () => {
  let requestTestWechatAuditConfig: jest.SpyInstance;
  const handleClickModifyFn = jest.fn();
  const customRender = (
    params: Omit<ConfigExtraButtonsProps, 'handleClickModify'>
  ) => {
    return superRender(
      <ConfigExtraButtons {...params} handleClickModify={handleClickModifyFn} />
    );
  };

  beforeEach(() => {
    mockUseCurrentUser();
    jest.useFakeTimers();
    requestTestWechatAuditConfig = system.testWechatAuditConfig();
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
    cleanup();
  });

  describe('render snap when hidden val is true', () => {
    it('render isConfigClosed is true', () => {
      const { baseElement } = customRender({
        isConfigClosed: true,
        extraButtonsVisible: true,
        enabled: false
      });
      expect(baseElement).toMatchSnapshot();
    });

    it('render extraButtonsVisible is false', () => {
      const { baseElement } = customRender({
        isConfigClosed: false,
        extraButtonsVisible: false,
        enabled: true
      });
      expect(baseElement).toMatchSnapshot();
    });

    it('render enabled is false', async () => {
      const { baseElement } = customRender({
        isConfigClosed: false,
        extraButtonsVisible: true,
        enabled: false
      });

      const systemIconBtn = getAllBySelector(
        '.system-config-button',
        baseElement
      );
      fireEvent.click(systemIconBtn[0]);
      await act(async () => jest.advanceTimersByTime(300));
    });
  });

  describe('render snap form', () => {
    it('render popover cancel btn', async () => {
      const { baseElement } = customRender({
        isConfigClosed: false,
        extraButtonsVisible: true,
        enabled: true
      });

      const btnPopoverOpen = getAllBySelector(
        '.system-config-button',
        baseElement
      )[0];
      fireEvent.click(btnPopoverOpen);
      await act(async () => jest.advanceTimersByTime(500));

      fireEvent.change(getBySelector('#wechatUserID', baseElement), {
        target: {
          value: '12345'
        }
      });

      expect(getBySelector('#wechatUserID', baseElement)).toHaveAttribute(
        'value',
        '12345'
      );

      expect(screen.getByText('取 消')).toBeInTheDocument();
      await act(async () => jest.advanceTimersByTime(300));
      expect(baseElement).toMatchSnapshot();

      fireEvent.click(screen.getByText('取 消'));

      fireEvent.click(btnPopoverOpen);
      expect(getBySelector('#wechatUserID', baseElement)).toHaveAttribute(
        'value',
        ''
      );
    });

    it('render form submit success', async () => {
      const { baseElement } = customRender({
        isConfigClosed: false,
        extraButtonsVisible: true,
        enabled: true
      });

      await act(async () => jest.advanceTimersByTime(3300));
      const btnPopoverOpen = getAllBySelector(
        '.system-config-button',
        baseElement
      )[0];
      fireEvent.click(btnPopoverOpen);
      await act(async () => jest.advanceTimersByTime(500));

      fireEvent.change(getBySelector('#wechatUserID', baseElement), {
        target: {
          value: '12345'
        }
      });
      await act(async () => jest.advanceTimersByTime(500));

      fireEvent.click(screen.getByText('确 认'));
      await act(async () => jest.advanceTimersByTime(500));
      expect(
        screen.getByText('正在向企业微信用户推送消息...')
      ).toBeInTheDocument();

      await act(async () => jest.advanceTimersByTime(2600));
      expect(requestTestWechatAuditConfig).toHaveBeenCalledTimes(1);
      expect(requestTestWechatAuditConfig).toHaveBeenCalledWith({
        wechat_id: '12345'
      });

      await act(async () => jest.advanceTimersByTime(400));
      expect(
        screen.getByText('已成功将消息推送至指定账号')
      ).toBeInTheDocument();
      await act(async () => jest.advanceTimersByTime(500));
      expect(baseElement).toMatchSnapshot();
    });

    it('render submit btn when api error', async () => {
      requestTestWechatAuditConfig.mockImplementation(() =>
        createSpySuccessResponse({
          data: {
            is_message_sent_normally: false,
            error_message: 'error info'
          }
        })
      );
      const { baseElement } = customRender({
        isConfigClosed: false,
        extraButtonsVisible: true,
        enabled: true
      });

      await act(async () => jest.advanceTimersByTime(3300));
      const btnPopoverOpen = getAllBySelector(
        '.system-config-button',
        baseElement
      )[0];
      fireEvent.click(btnPopoverOpen);
      await act(async () => jest.advanceTimersByTime(500));

      fireEvent.change(getBySelector('#wechatUserID', baseElement), {
        target: {
          value: '12345'
        }
      });
      await act(async () => jest.advanceTimersByTime(500));

      fireEvent.click(screen.getByText('确 认'));
      await act(async () => jest.advanceTimersByTime(500));
      expect(
        screen.getByText('正在向企业微信用户推送消息...')
      ).toBeInTheDocument();

      await act(async () => jest.advanceTimersByTime(2600));
      expect(requestTestWechatAuditConfig).toHaveBeenCalledTimes(1);
      expect(requestTestWechatAuditConfig).toHaveBeenCalledWith({
        wechat_id: '12345'
      });

      await act(async () => jest.advanceTimersByTime(400));

      expect(
        screen.queryByText('已成功将消息推送至指定账号')
      ).not.toBeInTheDocument();

      expect(screen.queryByText('error info')).toBeInTheDocument();
    });
  });
});
