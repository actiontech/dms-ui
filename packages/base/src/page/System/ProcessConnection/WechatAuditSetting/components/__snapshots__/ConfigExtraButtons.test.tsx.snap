// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base/System/ProcessConnection/WechatAuditSetting/ConfigExtraButtons render snap form render form submit success 1`] = `
<body>
  <div>
    <div
      class="ant-space ant-space-horizontal ant-space-align-center"
    >
      <div
        class="ant-space-item"
        style="margin-right: 12px;"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center basic-tooltips-wrapper css-4jbtl9"
        >
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn ant-btn-text ant-btn-icon-only ant-btn-loading system-config-button basic-button-wrapper css-geipcv"
              disabled=""
              type="button"
            >
              <span
                class="ant-btn-icon ant-btn-loading-icon"
              >
                <span
                  aria-label="loading"
                  class="anticon anticon-loading anticon-spin"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="loading"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="0 0 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="ant-space-item"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center basic-tooltips-wrapper css-4jbtl9"
        >
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn ant-btn-text ant-btn-icon-only system-config-button basic-button-wrapper css-geipcv"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  color="#292C33"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6.162 12.667H14V14H2v-2.829l6.6-6.6 2.828 2.83zm3.38-9.038 1.415-1.414a.667.667 0 0 1 .942 0L13.785 4.1a.667.667 0 0 1 0 .942l-1.414 1.414z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="ant-popover ant-popover-hidden ant-popover-placement-bottomRight"
      style="--arrow-x: 0px; --arrow-y: 0px; left: 0px; top: 0px; box-sizing: border-box; pointer-events: none;"
    >
      <div
        class="ant-popover-arrow"
        style="position: absolute;"
      />
      <div
        class="ant-popover-content"
      >
        <div
          class="ant-popover-inner"
          role="tooltip"
        >
          <div
            class="ant-popover-inner-content"
          >
            <div
              class="ant-space ant-space-vertical full-width-element"
            >
              <div
                class="ant-space-item"
                style="margin-bottom: 16px;"
              >
                <form
                  class="ant-form ant-form-horizontal"
                >
                  <div
                    class="ant-form-item css-1jlm9cy"
                    style="margin-bottom: 0px;"
                  >
                    <div
                      class="ant-row ant-form-item-row"
                    >
                      <div
                        class="ant-col ant-col-24 ant-form-item-label"
                      >
                        <label
                          class="ant-form-item-required ant-form-item-no-colon"
                          for="wechatUserID"
                          title="企业微信用户ID"
                        >
                          企业微信用户ID
                        </label>
                      </div>
                      <div
                        class="ant-col ant-col-24 ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              aria-required="true"
                              class="ant-input basic-input-wrapper css-1pd1cd5"
                              id="wechatUserID"
                              placeholder="请输入企业微信用户ID"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
              <div
                class="ant-space-item"
              >
                <div
                  class="ant-row"
                >
                  <div
                    class="ant-col ant-col-24"
                    style="text-align: right;"
                  >
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-right: 12px;"
                      >
                        <button
                          class="ant-btn ant-btn-default ant-btn-sm basic-button-wrapper css-geipcv"
                          type="button"
                        >
                          <span>
                            取 消
                          </span>
                        </button>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        <button
                          class="ant-btn ant-btn-primary ant-btn-sm basic-button-wrapper css-geipcv"
                          type="button"
                        >
                          <span>
                            确 认
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-message ant-message-top"
    style="left: 50%; transform: translateX(-50%); top: 8px;"
  >
    <div
      class="ant-message-notice ant-message-notice-success"
    >
      <div
        class="ant-message-notice-content"
      >
        <div
          class="ant-message-custom-content ant-message-success"
        >
          <span
            aria-label="check-circle"
            class="anticon anticon-check-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="check-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
              />
            </svg>
          </span>
          <span>
            已成功将消息推送至指定账号
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`base/System/ProcessConnection/WechatAuditSetting/ConfigExtraButtons render snap form render popover cancel btn 1`] = `
<body>
  <div>
    <div
      class="ant-space ant-space-horizontal ant-space-align-center"
    >
      <div
        class="ant-space-item"
        style="margin-right: 12px;"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center ant-popover-open basic-tooltips-wrapper css-4jbtl9"
        >
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn ant-btn-text ant-btn-icon-only system-config-button basic-button-wrapper css-geipcv"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  height="16"
                  viewBox="0 0 17 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.334 6.667H14l-6 8.666v-6H3.334l6-8.666z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="ant-space-item"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center basic-tooltips-wrapper css-4jbtl9"
        >
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn ant-btn-text ant-btn-icon-only system-config-button basic-button-wrapper css-geipcv"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  color="#292C33"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6.162 12.667H14V14H2v-2.829l6.6-6.6 2.828 2.83zm3.38-9.038 1.415-1.414a.667.667 0 0 1 .942 0L13.785 4.1a.667.667 0 0 1 0 .942l-1.414 1.414z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="ant-popover ant-popover-placement-bottomRight"
      style="--arrow-x: 0px; --arrow-y: 0px; left: -1000vw; top: -1000vh; box-sizing: border-box;"
    >
      <div
        class="ant-popover-arrow"
        style="position: absolute;"
      />
      <div
        class="ant-popover-content"
      >
        <div
          class="ant-popover-inner"
          role="tooltip"
        >
          <div
            class="ant-popover-inner-content"
          >
            <div
              class="ant-space ant-space-vertical full-width-element"
            >
              <div
                class="ant-space-item"
                style="margin-bottom: 16px;"
              >
                <form
                  class="ant-form ant-form-horizontal"
                >
                  <div
                    class="ant-form-item css-1jlm9cy ant-form-item-has-success"
                    style="margin-bottom: 0px;"
                  >
                    <div
                      class="ant-row ant-form-item-row"
                    >
                      <div
                        class="ant-col ant-col-24 ant-form-item-label"
                      >
                        <label
                          class="ant-form-item-required ant-form-item-no-colon"
                          for="wechatUserID"
                          title="企业微信用户ID"
                        >
                          企业微信用户ID
                        </label>
                      </div>
                      <div
                        class="ant-col ant-col-24 ant-form-item-control"
                      >
                        <div
                          class="ant-form-item-control-input"
                        >
                          <div
                            class="ant-form-item-control-input-content"
                          >
                            <input
                              aria-required="true"
                              class="ant-input ant-input-status-success basic-input-wrapper css-1pd1cd5"
                              id="wechatUserID"
                              placeholder="请输入企业微信用户ID"
                              type="text"
                              value="12345"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
              <div
                class="ant-space-item"
              >
                <div
                  class="ant-row"
                >
                  <div
                    class="ant-col ant-col-24"
                    style="text-align: right;"
                  >
                    <div
                      class="ant-space ant-space-horizontal ant-space-align-center"
                    >
                      <div
                        class="ant-space-item"
                        style="margin-right: 12px;"
                      >
                        <button
                          class="ant-btn ant-btn-default ant-btn-sm basic-button-wrapper css-geipcv"
                          type="button"
                        >
                          <span>
                            取 消
                          </span>
                        </button>
                      </div>
                      <div
                        class="ant-space-item"
                      >
                        <button
                          class="ant-btn ant-btn-primary ant-btn-sm basic-button-wrapper css-geipcv"
                          type="button"
                        >
                          <span>
                            确 认
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`base/System/ProcessConnection/WechatAuditSetting/ConfigExtraButtons render snap when hidden val is true render extraButtonsVisible is false 1`] = `
<body>
  <div>
    <div
      class="ant-space ant-space-horizontal ant-space-align-center"
      hidden=""
    >
      <div
        class="ant-space-item"
        style="margin-right: 12px;"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center basic-tooltips-wrapper css-4jbtl9"
        >
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn ant-btn-text ant-btn-icon-only system-config-button basic-button-wrapper css-geipcv"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  height="16"
                  viewBox="0 0 17 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.334 6.667H14l-6 8.666v-6H3.334l6-8.666z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="ant-space-item"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center basic-tooltips-wrapper css-4jbtl9"
        >
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn ant-btn-text ant-btn-icon-only system-config-button basic-button-wrapper css-geipcv"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  color="#292C33"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6.162 12.667H14V14H2v-2.829l6.6-6.6 2.828 2.83zm3.38-9.038 1.415-1.414a.667.667 0 0 1 .942 0L13.785 4.1a.667.667 0 0 1 0 .942l-1.414 1.414z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`base/System/ProcessConnection/WechatAuditSetting/ConfigExtraButtons render snap when hidden val is true render isConfigClosed is true 1`] = `
<body>
  <div>
    <div
      class="ant-space ant-space-horizontal ant-space-align-center"
      hidden=""
    >
      <div
        class="ant-space-item"
        style="margin-right: 12px;"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center basic-tooltips-wrapper css-4jbtl9"
        >
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn ant-btn-text ant-btn-icon-only system-config-button basic-button-wrapper css-geipcv"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  height="16"
                  viewBox="0 0 17 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M9.334 6.667H14l-6 8.666v-6H3.334l6-8.666z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="ant-space-item"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center basic-tooltips-wrapper css-4jbtl9"
        >
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn ant-btn-text ant-btn-icon-only system-config-button basic-button-wrapper css-geipcv"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  color="#292C33"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6.162 12.667H14V14H2v-2.829l6.6-6.6 2.828 2.83zm3.38-9.038 1.415-1.414a.667.667 0 0 1 .942 0L13.785 4.1a.667.667 0 0 1 0 .942l-1.414 1.414z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
